payment_integration_utils/payment_integration_utils/server_overrides/doctype/payment_entry.py

```
@@ -33,6 +33,25 @@ def onload(doc: PaymentEntry, method=None):
+
+
+def _transfer_method_options(doc: PaymentEntry) -> list[str] | None:
+    """First ``payment_transfer_method_options`` hook returning a non-empty list
+    narrows the PE's transfer-method choices to what the claiming integration
+    supports (e.g. a bank with no UPI/Link). ``None`` -> the full default set, so
+    an integration that registers nothing (or none installed) is unaffected. A
+    resolver error degrades to the full set rather than breaking the form."""
+    for path in frappe.get_hooks("payment_transfer_method_options"):
+        try:
+            methods = frappe.get_attr(path)(doc)
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)